lang/java/compiler/src/test/java/org/apache/avro/compiler/schema/TestSchemas.java

```diff
@@ -38,36 +59,11 @@
       + "                     {\"name\":\"methodName\",\"type\":{\"type\":\"string\",\"avro.java.string\":\"String\"}}\n"
       + "                  ]}},\n" + "              {\"name\":\"node\",\"type\":\"SampleNode\"}]}}}" + "]}";
 
-  private static class PrintingVisitor implements SchemaVisitor {
-
-    @Override
-    public SchemaVisitorAction visitTerminal(Schema terminal) {
-      System.out.println("Terminal: " + terminal.getFullName());
-      return SchemaVisitorAction.CONTINUE;
-    }
-
-    @Override
-    public SchemaVisitorAction visitNonTerminal(Schema terminal) {
-      System.out.println("NONTerminal start: " + terminal.getFullName());
-      return SchemaVisitorAction.CONTINUE;
-    }
-
-    @Override
-    public SchemaVisitorAction afterVisitNonTerminal(Schema terminal) {
-      System.out.println("NONTerminal end: " + terminal.getFullName());
-      return SchemaVisitorAction.CONTINUE;
-    }
-
-    @Override
-    public Object get() {
-      return null;
-    }
```

Comment:
I don't see much value in introducing Mockito for this. This visitor is very simple! It looks even simpler than the new Mockito based one.

Reply:
Yes, actually if we only want to confirm the execution order of the visitor, we can just spy on `CloningVisitor` and totaly get rid of the `printingVisitor`, do you think that will be better?

Reply:
One could also override the methods, print and call super:
```
CloningVisitor cv = new CloningVisitor(recSchema) {
  @Override
    public SchemaVisitorAction visitTerminal(Schema terminal) {
      System.out.println("Terminal: " + terminal.getFullName());
      return super.visitTerminal(terminal);
    }

   // ...
};
```

But if you need to do this at many places then it becomes ugly. Having around `PrintingVisitor` make is as easy as `Schemas.visit(recSchema, new PrintingVisitor());`

Reply:
Yes I agree with you. But just adding the print info doesn't effect the testing result, won't it better to explictly verify the execution times and order to enpower the test? Do you think the verification of execution order seems redundant in this case?